### DISCUSSION TITLE
A code style question

### DISCUSSION THREAD

**Discussion by wzq1022** [spring-projects/spring-kafka#3348] [General]
In `KafkaListenerEndpointRegistry` , line 399
 ```java
		if (listenerContainersToStop.size() > 0) {
			...
		}
```
why not use 
```java
		if (!listenerContainersToStop.isEmpty()) {
			...
		}
```
Normally, IDEA will prompt for modification, but why didn't the person who modified this file modify it? Is there any difference? I'm curious.

**Comment by artembilan** [comment]
I think such a suggestion from the IDEA has been introduced after that change had been made in this class - five years ago: https://github.com/spring-projects/spring-kafka/pull/1135.

So, there is no limit for perfection and that indeed has to be fixed as it is suggested: https://stackoverflow.com/questions/9341740/ifliststr-size-0-versus-ifliststr-isempty.

Converting this to issue and feel free to contribute the fix: https://github.com/spring-projects/spring-kafka/blob/main/CONTRIBUTING.adoc

**Reply by artembilan** [reply]
Here it is: https://github.com/spring-projects/spring-kafka/issues/3349

**Reply by wzq1022** [reply]
thanks for your reply.
